app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/ExposureNotificationsRepository.kt

```diff
@@ -311,14 +311,23 @@ class ExposureNotificationsRepository @Inject constructor(
                 accepted = false
             )
         })
-        val latestExposure = db.dao().getLatest().firstOrNull()?.daysSinceEpoch
-        val lastNotifiedExposure = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
-        if (latestExposure != null && latestExposure != lastNotifiedExposure) {
+
+        val latestExposure = db.dao().getLatest().firstOrNull()
+        val latestExposureTime = latestExposure?.daysSinceEpoch
+        val lastNotifiedExposureTime = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
+        val latestExposureTimestamp = latestExposure?.importTimestamp ?: 0L
+
+        val appOpenedTimestamp = prefs.getLastTimeAppOpened()
+        val appOpenedAfterLastNotification = latestExposureTimestamp < appOpenedTimestamp
+        val shouldShowNotification =
```

Comment:
Hodil by se taky komentář v kódu lidsky za jakých podmínek se zobrazuje notifikace. Musíme si tímto kódem být jistí, je to základní funkce eRoušky.